app/models/subscription.rb

```diff
@@ -35,19 +35,27 @@ def name
     end
   end
 
-  def subject_text(creation)
-    authors = if self.class.anonymous_creation?(creation)
-                "Anonymous"
-              else
-                creation.pseuds.map(&:byline).to_sentence
-              end
-    chapter_text = creation.is_a?(Chapter) ? "#{creation.chapter_header} of " : ""
-    work_title = creation.is_a?(Chapter) ? creation.work.title : creation.title
-    text = "#{authors} posted #{chapter_text}#{work_title}"
-    text += subscribable_type == "Series" ? " in the #{self.name} series" : ""
-  end
+  # Guard against scenarios that may break anonymity or other things.
+  # Emails should only contain works or chapters.
+  # Emails should only contain posted works or chapters.
+  # Emails should never contain chapters of draft works.
+  # Emails should never contain orphaned works or chapters.
+  # TODO: AO3-3620 & AO3-5696: Allow subscriptions to orphan_account to receive
+  # notifications.
+  # Emails for user subs should never contain anon works or chapters.
+  # Emails for work subs should never contain anything but chapters.
+  # Emails for subs to anon works or series should never contain non-anon works
+  # or chapters, or vice versa.
+  def valid_notification_entry?(creation)
+    return false unless creation.is_a?(Chapter) || creation.is_a?(Work)
+    return false unless creation.try(:posted)
+    return false if creation.is_a?(Chapter) && !creation.work.try(:posted)
+    return false if creation.pseuds.any? { |p| p.user == User.orphan_account }
+    return false if subscribable_type == "User" && creation.anonymous?
+    return false if subscribable_type == "Work" && !creation.is_a?(Chapter)
+    return false if subscribable.respond_to?(:anonymous?) &&
+                    subscribable.anonymous? != creation.anonymous?
```

Comment:
Are these checks new? If so, it'd be good to add some QA steps for them.

(I'm also not convinced that this is the right behavior for anonymous series. A series is marked as anonymous if any of the works in the series are anonymous, but until the behavior outlined [here](https://otwarchive.atlassian.net/browse/AO3-1250?focusedCommentId=333510) is implemented, users are still allowed to mix anonymous and non-anonymous works in a single series, so I think the notifications should reflect that. If you want to get a head start on AO3-1250, I suppose you could suppress all series notifications for changes to anonymous works, but this does the opposite.)

Reply:
They are new, yup, and I'll add steps make sure the "you shouldn't get emails" scenarios are covered in testing as well as the "you should get emails." 

And oh my god, I thought we'd prevented mixed anonymous/non-anonymous series _years_ ago. (I remember a specific volunteer who did a lot of testing around it and they haven't been around for... maybe six years or more?) Thank you for telling me it is still to-do.